src/url.cpp

```diff
@@ -308,24 +310,25 @@ namespace ada {
       bool is_forbidden{false};
       uint8_t ascii_runner{0};
 
+      buffer.reserve(input.size());
       std::transform(input.begin(), input.end(), std::back_inserter(buffer), [&is_forbidden, &ascii_runner](char c) -> char {
         is_forbidden |= ada::unicode::is_forbidden_domain_code_point(c);
         ascii_runner |= c;
         return (uint8_t((c|0x20) - 0x61) <= 25 ? (c|0x20) : c);}
       );
-      if(ascii_runner>=128) { goto slow_fallback; }
-      if(is_forbidden) { goto slow_fallback; } /* we may have a '%' */
-      if (buffer.find("xn-") == std::string_view::npos) {
+      if (ascii_runner < 128 && !is_forbidden && buffer.find("xn-") == std::string_view::npos) {
```

Comment:
As far as I know, the order of boolean expressions are important if you want to quit the execution flow earlier (for performance). Should we change the order of this line?